src/HttpHandler/Guzzle6HttpHandler.php

```diff
@@ -57,6 +68,37 @@ public function __invoke(RequestInterface $request, array $options = [])
      */
     public function async(RequestInterface $request, array $options = [])
     {
-        return $this->client->sendAsync($request, $options);
+        $requestEvent = null;
+
+        if ($this->logger) {
+            $requestEvent = new LogEvent();
+
+            $requestEvent->method = $request->getMethod();
+            $requestEvent->url = $request->getUri()->__toString();
+            $requestEvent->headers = $request->getHeaders();
+            $requestEvent->payload = $request->getBody()->getContents();
+            $requestEvent->retryAttempt = $options['retryAttempt'] ?? null;
+            $requestEvent->serviceName = $options['serviceName'];
+            $requestEvent->clientId = $options['clientId'];
+            $requestEvent->requestId = spl_object_id($request);
+
+            $this->logRequest($requestEvent);
+        }
+
+        return $this->client->sendAsync($request, $options)->then(function (ResponseInterface $response) use ($requestEvent) {
```

Comment:
nit, but this could be attached only when the logger is defined, which would look a little more cleaner. e.g.

```php
$promise = $this->client->sendAsync($request, $options);
if ($this->logger) {
    $promise->then( /** ... */);
}
return $promise;
```